### DISCUSSION TITLE
Add Chunk Load Information to API

### DISCUSSION THREAD

**Discussion by asdanjer** [PaperMC/Paper#10514] [Ideas]
### Is your feature request related to a problem?

Currently, the API does not have any way of getting information about the reason a chunk was loaded, by what (load ticket type), and at what level it was loaded (entity ticking, redstone ticking) (ticket level). This would be really useful information for debugging and administration purposes. Especially to quickly find what is causing lag, where, and who made it, that would be really useful. Currently, this is only possible with workarounds like checking around players, etc.

### Describe the solution you'd like.

Expose the ticket system in the API with capabilities to retrieve both the ticket level and type.

### Describe alternatives you've considered.

The alternative is to check all chunks around the player and then determine it through an after-the-fact calculation. However, this becomes quite complicated when there are multiple people and Load resons. Same for determining the load level by checking chunks on the edge of loaded areas. It is also not possible to distinguish, for example, chunk loaders from players except by checking the size of the continuously loaded area.

**Comment by Cryptite** [comment]
Partially-relevant patches we run for something to this effect on our fork:
https://github.com/Cryptite/Slice/blob/master/patches/server/0035-ChunkStatusChangeEvent.patch
https://github.com/Cryptite/Slice/blob/master/patches/api/0019-ChunkStatusChangeEvent.patch
